ServerCore/Pages/Pieces/Create.cshtml.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Pieces
+{
+    [Authorize(Policy = "IsAuthorOfPuzzle")]
```

Comment:
This should probably be IsEventAdminOrAuthorOfPuzzle -- we generally let event admins do anything.